content/post/gld.md

```diff
@@ -0,0 +1,1355 @@
+We treat each license as a TF-IDF-weighted bag-of-words - that is, as a set of words where every
+word has a weight corresponding to the frequency in the license text (term frequency) and
+throughout the whole database (document frequency). This is a proven approach for large scale
+similarity detection which we used multiple times in the past. Although 400 items is clearly
+not large scale at all, it still makes sense to employ LSH because of the O(1) complexity
+guarantee. We saw that it works reasonably well in practice and introduces the minimal overhead,
+say 20MB of memory for the hashes and the vocabulary. The hashing algorithm which we use is
+[Weighted MinHash](http://static.googleusercontent.com/media/research.google.com/en//pubs/archive/36928.pdf)
+- again, battle-tested in the past, e.g. in [Apollo](https://github.com/src-d/apollo) or
+[bags deduplication](https://blog.sourced.tech/post/minhashcuda/).
+We set 75% Jaccard similarity threshold and hash length 154
+after careful tuning of the false positive vs. false negative vs. performance trade-off.
+Since we discard the text structure by treating sequences as sets, we further calculate the Levenshtein
+distance to the matched database records in order to determine the precise confidence value.
+
+We look at the README file if the analyzed project does not contain a license file. This happens
+in more than 7% of the cases in the 1k dataset and x% in Public Git Archive (182,000 repositories).
+There is a fair chance that the license name is mentioned in the README, so we apply
+[Named Entity Recognition](https://en.wikipedia.org/wiki/Named-entity_recognition)
+to find them using the excellent [jdkato/prose](https://github.com/jdkato/prose) NLP library for Go.
+
+Unfortunately, the devil is not in the core algorithm but rather in numerous details which cover
+the huge variance of the conditions.
+
+## File names
+
+There are many possible license file names. A few examples:
+
+```nohighlight
+LICENSE
+license.txt
+License.md
+lisence.html
+lisense.rst
+copyright
+copying
+MIT.txt
+gpl-2.0
+legal
+```
+
+All of them may contain useful information, so we have to design a suitable regular expression for
+them, which is currently
+
+```go
+var (
+	licenseFileNames = []string{
+		"li[cs]en[cs]e(s?)",
+		"legal",
+		"copy(left|right|ing)",
+		"unlicense",
+		"l?gpl([-_ v]?)(\\d\\.?\\d)?",
+		"bsd",
+		"mit",
+		"apache",
+	}
+	licenseFileRe = regexp.MustCompile(
+		fmt.Sprintf("^(|.*[-_. ])(%s)(|[-_. ].*)$",
+			strings.Join(licenseFileNames, "|")))
+)
+```
+
+There may be also directories which are named like a license file, and we need to look inside.
+A few projects contained symbolic links to the actual license texts, and we need to resolve them.
+One project even has the license file name as the only content of `LICENSE` and we treat such files
```

Comment:
I don't understand this sentence. Do you mean it has the license file inside of a directory named `LICENSE`
or that a file named `LICENSE` exists but the only content is the path to another license description? 

Please, clarify in the text.